File: backend/app/schemas/case_brain.py
"""
Pydantic schemas for the Case Brain timeline API.
"""

from __future__ import annotations

import uuid
from typing import Optional

from pydantic import BaseModel


class CaseBrainLogEntry(BaseModel):
    """A single entry in the Case Brain timeline."""

    brain_entry_id: int
    email_id: Optional[uuid.UUID]
    occurred_at: str
    source_type: str
    source_reference: Optional[str]
    source_actor: Optional[str]
    update_summary: str
    logged_by: Optional[str]


class CaseBrainTimelineResponse(BaseModel):
    """Timeline response for a Matter's Case Brain log."""

    matter_key: str
    total: int
    entries: list[CaseBrainLogEntry]

File: frontend/src/types/caseBrain.ts
export interface CaseBrainLogEntry {
  brain_entry_id: number;
  email_id: string | null;
  occurred_at: string;
  source_type: string;
  source_reference: string | null;
  source_actor: string | null;
  update_summary: string;
  logged_by: string | null;
}

export interface CaseBrainTimelineResponse {
  matter_key: string;
  total: number;
  entries: CaseBrainLogEntry[];
}
